public/app/features/explore/Logs/LogsContainer.tsx

```diff
@@ -59,45 +60,75 @@ interface LogsContainerProps extends PropsFromRedux {
 
 interface LogsContainerState {
   logDetailsFilterAvailable: boolean;
+  /**
+   * LogContext support for mixed data sources.
+   * @alpha
+   */
+  logContextSupport: Record<string, DataSourceApi<DataQuery> & DataSourceWithLogsContextSupport<DataQuery>>;
 }
 
 class LogsContainer extends PureComponent<LogsContainerProps, LogsContainerState> {
   state: LogsContainerState = {
     logDetailsFilterAvailable: false,
+    logContextSupport: {},
   };
 
   componentDidMount() {
-    this.checkFiltersAvailability();
+    this.checkDataSourcesFeatures();
   }
 
   componentDidUpdate(prevProps: LogsContainerProps) {
-    this.checkFiltersAvailability();
+    this.checkDataSourcesFeatures();
   }
 
-  private checkFiltersAvailability() {
+  private checkDataSourcesFeatures() {
     const { logsQueries, datasourceInstance } = this.props;
 
-    if (!logsQueries) {
+    if (!logsQueries || !datasourceInstance) {
       return;
     }
 
-    if (datasourceInstance?.modifyQuery || hasToggleableQueryFiltersSupport(datasourceInstance)) {
-      this.setState({ logDetailsFilterAvailable: true });
+    let newState: LogsContainerState = { ...this.state, logDetailsFilterAvailable: false };
+
+    // Not in mixed mode.
+    if (datasourceInstance.uid !== MIXED_DATASOURCE_NAME) {
+      if (datasourceInstance?.modifyQuery || hasToggleableQueryFiltersSupport(datasourceInstance)) {
+        newState.logDetailsFilterAvailable = true;
+      }
+      if (hasLogsContextSupport(datasourceInstance)) {
+        logsQueries.forEach(({ refId }) => {
+          newState.logContextSupport[refId] = datasourceInstance;
+        });
+      }
+      this.setState(newState);
       return;
     }
 
-    const promises = [];
+    // Mixed mode.
+    const dsPromises: Array<Promise<{ ds: DataSourceApi, refId: string}>> = [];
     for (const query of logsQueries) {
-      if (query.datasource) {
-        promises.push(getDataSourceSrv().get(query.datasource));
+      if (query.datasource && !newState.logContextSupport[query.refId]) {
+        dsPromises.push(new Promise((resolve) => {
+          getDataSourceSrv().get(query.datasource).then(ds => {
+            resolve({ ds, refId: query.refId });
+          })
+        }));
       }
     }
 
-    Promise.all(promises).then((dataSources) => {
-      const logDetailsFilterAvailable = dataSources.some(
-        (ds) => ds.modifyQuery || hasToggleableQueryFiltersSupport(ds)
-      );
-      this.setState({ logDetailsFilterAvailable });
+    if (!dsPromises.length) {
+      return;
```

Comment:
I am just wondering if there are ever cases where we need to "reset" the state in cases there are no `dsPromises`? Aka setting every `logContextSupport[refId]` to false here?

Reply:
True. Added to line 126, along with checking the uid on line 106.